pkg/controller/job/job_controller.go

```diff
@@ -741,7 +751,7 @@ func (jm *Controller) syncJob(ctx context.Context, key string) (forget bool, rEr
 
 	activePods := controller.FilterActivePods(pods)
 	active := int32(len(activePods))
-	succeeded, failed := getStatus(&job, pods, uncounted, expectedRmFinalizers)
+	succeeded, failed, succeededPods, failedPods := getStatus(&job, pods, uncounted, expectedRmFinalizers)
```

Comment:
I would suggest to only return `succeededPods` and `failedPods` here. `succeeded` and `failed` can be computed outside.